src/sgdnet.cpp

```diff
@@ -261,64 +73,60 @@ inline void AdaptiveTranspose(arma::mat& x) {
 template <typename T>
 Rcpp::List SetupSgdnet(T&                   x,
                        std::vector<double>& y,
-                       const bool           is_sparse,
-                       const Rcpp::List&    control) {
-
-  const std::string   family_choice    = Rcpp::as<std::string>(control["family"]);
-  const bool          fit_intercept    = Rcpp::as<bool>(control["intercept"]);
-  const double        elasticnet_mix   = Rcpp::as<double>(control["elasticnet_mix"]);
-  std::vector<double> lambda           = Rcpp::as< std::vector<double> >(control["lambda"]);
-  const std::size_t   n_lambda         = Rcpp::as<std::size_t>(control["n_lambda"]);
-  const double        lambda_min_ratio = Rcpp::as<double>(control["lambda_min_ratio"]);
-  const bool          normalize        = Rcpp::as<bool>(control["normalize"]);
-  std::size_t         max_iter         = Rcpp::as<std::size_t>(control["max_iter"]);
-  const double        tol              = Rcpp::as<double>(control["tol"]);
-  const bool          debug            = Rcpp::as<bool>(control["debug"]);
-
-  std::size_t n_samples  = x.n_rows;
-  std::size_t n_features = x.n_cols;
+                       const bool          is_sparse,
+                       const Rcpp::List&   control) {
+  using namespace std;
+  using namespace sgdnet;
+
+  const string   family_choice    = Rcpp::as<string>(control["family"]);
+  const bool     fit_intercept    = Rcpp::as<bool>(control["intercept"]);
+  const double   elasticnet_mix   = Rcpp::as<double>(control["elasticnet_mix"]);
+  vector<double> lambda           = Rcpp::as<vector<double>>(control["lambda"]);
+  const unsigned n_lambda         = Rcpp::as<unsigned>(control["n_lambda"]);
+  const double   lambda_min_ratio = Rcpp::as<double>(control["lambda_min_ratio"]);
+  const bool     standardize      = Rcpp::as<bool>(control["normalize"]);
+  const unsigned max_iter         = Rcpp::as<unsigned>(control["max_iter"]);
+  const double   tol              = Rcpp::as<double>(control["tol"]);
+  const bool     debug            = Rcpp::as<bool>(control["debug"]);
+
+  unsigned n_samples  = x.rows();
+  unsigned n_features = x.cols();
```

Comment:
Isn't `unsigned` a type _modifier_? I'd expect `unsigned int` as the type here

Reply:
It is an alias for `unsigned int`. I am using simply to avoid verbosity but could change it if you wish.

Reply:
It looks like the standard guarantees `unsigned` is interpreted as `unsigned int` so don't worry about it: https://en.cppreference.com/w/cpp/language/types